src/App.tsx

```diff
@@ -1,26 +1,234 @@
+  const [controlChecked, setControlChecked] = useState<number[]>([]);
+  const [filter, setFilter] = useState('');
+  const inputRef = useRef<HTMLInputElement>(null);
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [loadingNewItem, setLoadingNewItem] = useState(false);
+  const [arrTodos, setArrTodos] = useState<number[]>([]);
+  const [delLoader, setDelLoader] = useState<number | null>(null);
+
+
+  useEffect(() => {
+    todosService
+      .getTodos()
+      .then(setTodoItem)
+      .catch(() => setStateError('Unable to load todos'));
+  }, []);
+
+  useEffect(() => {
+    if (errorState) {
+      const timer = setTimeout(() => {
+        setStateError(''); // Clear the error message after 3 seconds
+      }, 3000);
+
+      return () => clearTimeout(timer); // Clean up timer if component unmounts
+    }
+  }, [errorState]);
+
+
+  const getFilteredTodos = () => {
+    if (filter === 'active') {
+      return todoItem.filter(todo => !todo.completed);
+    }
+
+    if (filter === 'completed') {
+      return todoItem.filter(todo => todo.completed);
+    }
+
+    return todoItem;
+  };
+
+  const filteredTodos = getFilteredTodos();
+
+  const handleForm = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (creatNewTodos.trim() === '') {
+      return;
+    }
+
+    const newTodo: Omit<Todo, 'id'> = {
+      userId,
+      title: creatNewTodos,
+      completed: false,
+    };
+
+    const tempTodoItem = { ...newTodo, id: Date.now() };
+
+    // Add the new todo to the UI immediately (optimistic update)
+    setTodoItem(prev => [...prev, tempTodoItem]);
+    setArrTodos(prevItem => [...prevItem, tempTodoItem.id]);
+    setLoadingNewItem(true);
+
+    // Make API call to create the todo
+    todosService
+      .createPost(newTodo)
+      .then(createdTodo => {
+        // Update the UI with the created todo only if the API call succeeds
+        setTodoItem(prev =>
+          prev.map(todo => (todo.id === tempTodoItem.id ? createdTodo : todo)),
+        );
+        setCreateNewTodos('');
+        setArrTodos(prevItem =>
+          prevItem.map(id => (id === tempTodoItem.id ? createdTodo.id : id)),
+        );
+      })
+      .catch(() => {
+        setStateError('Unable to add a todo');
+      })
+      .finally(() => {
+        setLoadingNewItem(false);
+        // Remove the temporary item after the API call completes
+        setTimeout(() => {
+          setArrTodos([]); // Clear the temporary array after 1 second
+        }, 1000);
+      });
+  };
+
+  const forClearCompleted = () => {
+    const completedTodo = todoItem.filter(todo => todo.completed);
+    const completedIds = completedTodo.map(todo => todo.id);
+
+    setTodoItem(prev => prev.filter(todo => !completedIds.includes(todo.id)));
+
+    completedIds.forEach(id => {
+      todosService.deleteTodos(id).catch(() => {
+        setTimeout(() => setStateError(''), 3000);
+      });
+    });
+  };
+
+  function errorGetTodos() {
```

Comment:
It is important to stick to one way of creating functions, fix it everywhere
```suggestion
  const errorGetTodos = () => {
```